core/builder/src/main/java/io/quarkus/builder/JsonReader.java

```diff
@@ -0,0 +1,442 @@
+package io.quarkus.builder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * A json format reader.
+ * It follows the <a href="https://www.json.org/json-en.html>ECMA-404 The JSON Data Interchange Standard.</a>.
+ */
+public class JsonReader {
```

Comment:
I just know that this class is going to be misused to add more json stuff....

Reply:
That can of worms was open when the `Json` class was added to generate build output metrics, which Foivos' then reused to generate the json config output for native image :)